### DISCUSSION TITLE
Install instructions in docs and question about teams integration

### DISCUSSION THREAD

**Discussion by zerocarbthirty** [jertel/elastalert2#125] [Q&A] [answered]
Hello,

Three questions:

1)
Are the installation instructions for elastalert 2 correct? 

Is this not for the original elastalert?

from [](https://elastalert2.readthedocs.io/en/latest/running_elastalert.html)

$ pip install elastalert
or you can clone the ElastAlert repository for the most recent changes:

$ git clone https://github.com/Yelp/elastalert.git

2) is there a way to have the original and elastalert2 installed at the same time?

3) Has the Microsoft Teams integration been fixed or does it still send the notifications to Teams with no formatting?

**Comment by nsano-rururu** [comment]
It means that there are some parts that cannot be corrected in the document. Thank you for contacting us. to correct.

>Has the Microsoft Teams integration been fixed or does it still send the notifications to Teams with no formatting?

What do you mean? .. I do not understand. I haven't fixed it

**Reply by nsano-rururu** [reply]
>is there a way to have the original and elastalert2 installed at the same time?

I don't know the need.
Since some libraries have different versions to install, it is better not to install in the same python virtual environment as the original.

**Reply by nsano-rururu** [reply]
Is it better to switch to pymsteams or who wrote in the yelp / elastalert issue?

**Reply by nsano-rururu** [reply]
If you are familiar with pymsteams, you can make a pull request
Shouldn't it be? .. Or can't you?

**Comment by jertel** [comment, marked as answer]
1. I just pushed a fix for this. Thank you for pointing out the instructions were still pointing to the old repo.
2. You probably could do this but you might need to configure one of the ElastAlert's to use a different set of indices, if you plan on running them against the same Elasticsearch cluster. But I am also curious what the scenario is that calls for both to run at the same time. Are you trying to compare them?
3. I haven't used the Team's integration. If the bug is still present and you are able to correct it then a PR would be appreciated!

**Reply by nsano-rururu** [reply]
@jertel 

As a side note, there are no bugs in the implementation of alert notifications for ms teams. zerocarbthirty made the following comment as an issue.

https://github.com/Yelp/elastalert/issues/2727#issuecomment-761090363
>zerocarbthirty commented on 16 Jan
>There is also this python library that handles the formatting for messages sent to Teams webhooks.
>https://pypi.org/project/pymsteams/

**Comment by zerocarbthirty** [comment]
Yes it just formats it in a way that makes the notifications unreadable by
humans.

**Reply by nsano-rururu** [reply]
>Yes it just formats it in a way that makes the notifications unreadable by
humans.

so what?

**Reply by nsano-rururu** [reply]
>Has the Microsoft Teams integration been fixed or does it still send the notifications to Teams with no formatting?

I'm not interested in teams. You should make a pull request if you want to fix it.

**Reply by nsano-rururu** [reply]
Open source is not the responsibility of the developer to add features or fix bugs. So the developer is not obliged to meet your needs. Don't get me wrong. It's not commercial software.
